test/Dialect/FIRRTL/imdce.mlir

```diff
@@ -180,3 +180,52 @@ firrtl.circuit "ForwardConstant" {
     firrtl.strictconnect %zero, %sub_zero : !firrtl.uint<1>
   }
 }
+
+// -----
+
+firrtl.circuit "RefPorts" {
+  // CHECK-NOT: @dead_ref_writeonly
+  firrtl.module private @dead_ref_writeonly(in %source: !firrtl.uint<1>, out %dest: !firrtl.ref<uint<1>>) {
+    %dead_node = firrtl.node %source: !firrtl.uint<1>
+    %not = firrtl.not %dead_node : (!firrtl.uint<1>) -> !firrtl.uint<1>
+    %ref_not = firrtl.ref.send %not : !firrtl.uint<1>
+    firrtl.connect %dest, %ref_not : !firrtl.ref<uint<1>>, !firrtl.ref<uint<1>>
+  }
+
+  // CHECK-NOT: @dead_ref_module
+  firrtl.module private @dead_ref_module(in %source: !firrtl.uint<1>, out %dest: !firrtl.ref<uint<1>>) {
+  }
```

Comment:
I wonder if this should be legal. This is something that seems like should trip initialization checking in `ExpandWhens`.

Reply:
Oh, good call.  I'll drop this, also not especially useful.